### DISCUSSION TITLE
How to calibrate Illuminance in a presence sensor (illuminance_lux_calibration parameter)

### DISCUSSION THREAD

**Discussion by acseven** [Koenkk/zigbee2mqtt#20885] [Devices]
Hi,
I'm using a [TuYa ZY-M100-24G](https://www.zigbee2mqtt.io/devices/ZY-M100-24G.html#tuya-zy-m100-24g) presence sensor on my Z2M setup, which for the most part is working fine. 

However, illuminance values are not; this is a common issue I found discussed online but with unfortunately found no solutions. 

In any case, the device has a specific setting (`illuminance_lux_calibration`) that is apparently supposed to control the calibration of the light metering:

```
Illuminance lux calibration
Calibrates the illuminance_lux value (percentual offset), takes into effect on next report of device.
```

![image](https://github.com/Koenkk/zigbee2mqtt/assets/1176371/e00f8cb5-a584-441c-a922-91fe246abed1)

The issue is that I found no documentation that helps to understand to properly calibrate it. Can anyone shed some light on this?

Thank you

**Comment by sjorge** [comment]
Like all calibration settings it's an offset from reported.

Does it report 100lx but it's actually 50lx, you can set the calibration to -50, does it report 200lx but it's 500lx, your offset would be 300.

It doesn't work to well for logarithmic scaling values like lx from my experience.

**Reply by acseven** [reply]
Thanks for the clarification. In my case it does appear to have that scaling issue, as an offset value doesn't report the actual lx value (the value on the rightmost part of the graph should measure 190, while reporting 20):

![image](https://github.com/Koenkk/zigbee2mqtt/assets/1176371/9afa3139-eeef-4819-ab69-fd5c5076766c)

**Reply by sjorge** [reply]
@Koenkk could it be something got broken with one of the refactors? Or are those values applied pre illuminance -> illuminance_lux conversion ?

**Reply by Koenkk** [reply]
@sjorge they are applied after that: https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/src/index.ts#L227

**Reply by acseven** [reply]
Not knowing how this relates to Z2M, but this bug fix of illuminance measured values has been added to ZHA in HA 2024.1.6:
https://github.com/home-assistant/core/issues/23290
